### PR TITLE
airtanks and breathing tube in armor slots

### PR DESCRIPTION
**Pull request by nonfreegithub**
in MineClone/MineClonia it can be used in armor slot

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -144,11 +147,19 @@ local function register_air_tank(name, desc, color, uses, material)
 		inventory_image = "airtanks_airtank.png^[colorize:"..color.."^[mask:airtanks_airtank.png",
 		wield_image = "airtanks_airtank.png^[colorize:"..color.."^[mask:airtanks_airtank.png",
 		stack_max = 1,
+		_mcl_armor_element = "torso",
+		_mcl_armor_texture = "airtanks_chestplate_tank.png",
 	
 		on_place = function(itemstack, user, pointed_thing)
-			return use_airtank(itemstack, user)
+		   if minetest.get_modpath("mcl_armor") then
```
Comment (SmallJoker) [inline init.lua]: Inconsistent indent levels. This mod uses tabs to indent.

EDIT: Also for L208+ and any other occurrences

File: init.lua [hunk 2]
```diff
@@ -144,11 +147,19 @@ local function register_air_tank(name, desc, color, uses, material)
...
+		      mcl_armor.equip_on_use(itemstack, user, pointed_thing)
```
Comment (SmallJoker) [inline init.lua]: Is there no return value to forward to `on_place`?

EDIT: Also for L208+ and any other occurrences

File: init.lua [hunk 3]
```diff
@@ -608,7 +642,7 @@ local function tool_active(player, item)
 	local inv = player:get_inventory()
 	local hotbar = player:hud_get_hotbar_itemcount()
 	for i=1, hotbar do
-		if inv:get_stack("main", i):get_name() == item then
+		if inv:get_stack("armor", i):get_name() == item then
```
Comment (SmallJoker) [inline init.lua]: This slot is not guaranteed to exist, hence breaks games without armor slots.

File: init.lua (hunk 2 above)
Comment (nonfreegithub) [reply init.lua]: fixed

File: init.lua (hunk 3 above)
Comment (nonfreegithub) [reply init.lua]: fixed

File: init.lua (hunk 1 above)
Comment (nonfreegithub) [reply init.lua]: dunno how I can fix that, mi editor sets that ident

Comment (SmallJoker) [reply init.lua]: In that case, I would recommend you to check for editor settings to either automatically detect indent levels or to switch between them by hotkeys or a few clicks.

Comment (nonfreegithub) [reply init.lua]: fixed

File: init.lua [hunk 4]
```diff
@@ -243,11 +262,19 @@ local function register_air_tank_3(name, desc, color, uses, material)
...
 		on_place = function(itemstack, user, pointed_thing)
-			return use_airtank(itemstack, user)
+		   if minetest.get_modpath("mcl_armor") then
+		      mcl_armor.equip_on_use(itemstack, user, pointed_thing)
+		   else
+		      return use_airtank(itemstack, user)
+		   end
 		end,
 
+		on_secondary_use = minetest.get_modpath("mcl_armor") and mcl_armor.equip_on_use,
```
Comment (SmallJoker) [inline init.lua]: `mcl_armor` must be in mod.conf to guarantee that it is loaded before this mod. Internal mcl dependency trees might already do that but it's better to be safe than sorry.

That would also allow you to cache `minetest.get_modpath("mcl_armor")` in a separate variable like `local HAVE_MCL_ARMOR`.

Comment (nonfreegithub) [reply init.lua]: done

**Comment by SmallJoker** [comment]
Thanks. Will merge in a few days unless there are objections.

**Pull request merged**